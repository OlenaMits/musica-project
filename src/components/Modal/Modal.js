import React from "react";
import PropTypes from 'prop-types';
import { ReactComponent as CrossIcon } from './images/cross.svg';

import "./Modal.scss";

const Modal = (props) => {
    const { 
        show, 
        handleClose, 
        header, 
        closeButton, 
        text, 
        actions, 
        backgroundColor 
    } = props;

    return(
        <>
            <div className={`modal ${show ? "display-block" : "display-none"}`} tabIndex="-1" id="myModal">
                <div className="modal-dialog">
                    <div className="modal-content" style={{ backgroundColor }}>
                        <div className="modal-header">
                            <h5 className="modal-title">{header}</h5>
                            {closeButton && (
                                <button
                                    type="button" 
                                    className="btn-close" 
                                    onClick={handleClose}
                                >
                                    <CrossIcon/>
                                </button>
                            )}
                        </div>
                        <div className="modal-body">
                            <p>{text}</p>
                            <div className="btn-group">
                            {actions}
                            </div>
                        </div>         
                    </div>
                </div>
            </div>
            <div 
                className={`modal-backdrop ${show ? "display-block" : "display-none"}`} 
                onClick={handleClose}
            />
        </>
    )
};

Modal.propTypes = {
    show: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    closeButton: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
    actions: PropTypes.element.isRequired,
};

Modal.defaultProps = {
    header: 'Add to Cart',
    backgroundColor: '#aedaa6',
};

export default Modal;