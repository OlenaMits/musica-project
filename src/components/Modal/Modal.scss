.modal {
    position: fixed;
    top: 0;
    left: 0;
    color: #ffffff;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
    z-index: 2;

        &-content {
            position:fixed;
            width: 516px;
            width: 516px;
            height: auto;
            top:50%;
            left:50%;
            transform: translate(-50%,-50%);
            border-radius: 4px;
        }

        &-header {
            display: flex;
            justify-content: space-between;
            align-items: center;
            padding: 0 20px;
            height: 68px;
            border-radius: 4px;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
            font-size: 22px;
            line-height: 60px;
            font-weight: 500;
            font-family: "Helvetica Neue";
            background-color: rgba(0, 0, 0, 0.08);

            .btn-close {
                background: transparent;
                font-size: 20px;
                border: none;
                cursor: pointer;
                display: flex;
            }
        }

        &-body {
            height: 184px;
            display: flex;
            flex-direction: column;
            justify-content: space-between;
            align-items: center;
            padding: 1em;
            box-sizing: border-box;

            &-text {
                display: flex;
                flex-direction: column;
                justify-content: center;
                align-items: center;
            }
            
            .btn-group {
                display: flex;
                justify-content: center;
            }
        }
}

.modal-backdrop {
    width: 100vw;
    height: 100vh;
    position: fixed;
    top: 0;
    left: 0;
    background: rgba(0, 0, 0, 0.6);
    top: 0;
    z-index: 1;
}

.display-block {
    display: block;
}

.display-none {
    display: none;
}