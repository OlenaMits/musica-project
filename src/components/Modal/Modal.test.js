import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Modal from "./Modal";

describe('Modal tests', () => {
    const handleCloseMock = jest.fn();
    
    const baseProps = {
        closeButton: true,
        handleClose: handleCloseMock,
        backgroundColor: '#aedaa6',
        header: 'test',
        actions: <div/>,
    }

    it('should match snapshot modal hide', () => {
        const {asFragment} = render(
            <Modal 
                show={false} 
                text="modal hide"
                {...baseProps}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
   
    it('should match snapshot modal visible', () => {
        const {asFragment} = render(
            <Modal 
                show={true} 
                text="modal visible"
                {...baseProps}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
   
    it('should trigger action', () => {
        render(
            <Modal 
                show={true} 
                text="modal visible"
                {...baseProps}
            />);
        fireEvent.click(screen.getByTestId('modal-close-btn'));
        expect(handleCloseMock).toHaveBeenCalled();
    });
})