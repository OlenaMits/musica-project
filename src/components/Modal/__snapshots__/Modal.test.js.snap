// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests should match snapshot modal hide 1`] = `
<DocumentFragment>
  <div
    class="modal display-none"
    id="myModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
        style="background-color: rgb(174, 218, 166);"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            test
          </h5>
          <button
            class="btn-close"
            type="button"
          >
            <svg>
              cross.svg
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            modal hide
          </p>
          <div
            class="btn-group"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop display-none"
    data-testid="modal-close-btn"
  />
</DocumentFragment>
`;

exports[`Modal tests should match snapshot modal visible 1`] = `
<DocumentFragment>
  <div
    class="modal display-block"
    id="myModal"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
        style="background-color: rgb(174, 218, 166);"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
          >
            test
          </h5>
          <button
            class="btn-close"
            type="button"
          >
            <svg>
              cross.svg
            </svg>
          </button>
        </div>
        <div
          class="modal-body"
        >
          <p>
            modal visible
          </p>
          <div
            class="btn-group"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop display-block"
    data-testid="modal-close-btn"
  />
</DocumentFragment>
`;
